### DISCUSSION TITLE
Loading songs outside of SuperTux Source

### DISCUSSION THREAD

**Discussion by AM7999** [SuperTux/supertux#2217] [Q&A] [answered]
Hey i was just wondering how i would import songs that were not compiled with SuperTux Or if i could even import songs that were not compiled w/ SuperTux

**Comment by Semphriss** [comment, marked as answer]
You can!
1. Convert your music files to .ogg files (SuperTux only supports .ogg and .wav files, and .ogg files are usually preferable)
2. Place your .ogg's somewhere in your userdata directory (the same folder that contains the "config" files, the "profiles" folder, the "addons" folder, etc). You can place in in a new folder called "music", or alongside your levels.
3. Add a new file called `yourmusicname.music` alongside your .ogg and open it with any text editor, and put this in it:
```
(supertux-music
  (file "yourmusicfile.ogg")
)
```
(Adapt the file names as needed)
4. Optionally, you can add another line before the last line, like `  (loop-at 1.234)` where 1.234 is the number of seconds the song should jump to when it loops. This is useful if your song has an "intro" which should be skipped when looping. (fortress.music does that)
5. Optionally again, you can add yet another line like `  (loop-begin 56.789)` where 56.789 is where your song should start looping. That is useful if you want to cut out an "outro" from the song file.
6. From the editor, in your level, change your music and find the music file from your userdata folder, and select the file you created that ends with `.music`.

Hope it helps!

**Reply by AM7999** [reply]
Thanks!

**Comment by Linusintro** [comment]
Cool
